pkg/kubelet/cm/dra/manager.go

```diff
@@ -185,70 +171,75 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod, container *v1.Container) (*C
 				continue
 			}
 
-			resource := m.resources.get(claimName, pod.Namespace)
-			if resource == nil {
+			claimInfo := m.cache.get(claimName, pod.Namespace)
+			if claimInfo == nil {
 				return nil, fmt.Errorf("unable to get resource for namespace: %s, claim: %s", pod.Namespace, claimName)
 			}
 
-			klog.V(3).InfoS("add resource annotations", "claim", claimName, "annotations", resource.annotations)
-			annotations = append(annotations, resource.annotations...)
+			klog.V(3).InfoS("add resource annotations", "claim", claimName, "annotations", claimInfo.annotations)
+			annotations = append(annotations, claimInfo.annotations...)
 		}
 	}
 
 	return &ContainerInfo{Annotations: annotations}, nil
 }
 
-func (m *ManagerImpl) UnprepareResources(pod *v1.Pod) error {
-	m.Lock()
-	defer m.Unlock()
+// PrepareResources calls plugin NodePrepareResource from the registered DRA resource plugins.
+func (m *ManagerImpl) PrepareResources(pod *v1.Pod, container *v1.Container) (*ContainerInfo, error) {
+	if err := m.prepareContainerResources(pod, container); err != nil {
+		return nil, err
+	}
+	return m.getContainerInfo(pod, container)
+}
 
+func (m *ManagerImpl) UnprepareResources(pod *v1.Pod) error {
 	// Call NodeUnprepareResource RPC for every resource claim referenced by the pod
 	for _, podResourceClaim := range pod.Spec.ResourceClaims {
 		claimName := resourceclaim.Name(pod, &podResourceClaim)
 
-		resource := m.resources.get(claimName, pod.Namespace)
-		if resource == nil {
-			return fmt.Errorf("failed to get resource for namespace %s, claim %s", pod.Namespace, claimName)
+		claimInfo := m.cache.get(claimName, pod.Namespace)
+		if claimInfo == nil {
+			// skip calling NodeUnprepareResource if claim info is not cached
```

Comment:
nit: capitalize this comment and move it above the if

Reply:
done